OpenRA.Game/InstalledMods.cs

```diff
@@ -0,0 +1,111 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using OpenRA.FileSystem;
+using OpenRA.Primitives;
+
+namespace OpenRA
+{
+	public class InstalledMods : IReadOnlyDictionary<string, Manifest>
+	{
+		readonly Dictionary<string, Manifest> mods;
+
+		public InstalledMods()
+		{
+			mods = GetInstalledMods();
+		}
+
+		static IEnumerable<Pair<string, string>> GetCandidateMods()
+		{
+			// Get mods that are in the game folder.
+			var basePath = Platform.ResolvePath(Path.Combine(".", "mods"));
+			var mods = Directory.GetDirectories(basePath)
+				.Select(x => Pair.New(x.Substring(basePath.Length + 1), x))
+				.ToList();
+
+			foreach (var m in Directory.GetFiles(basePath, "*.oramod"))
+				mods.Add(Pair.New(Path.GetFileNameWithoutExtension(m), m));
+
+			// Get mods that are in the support folder.
+			var supportPath = Platform.ResolvePath(Path.Combine("^", "mods"));
+			if (!Directory.Exists(supportPath))
+				return mods;
+
+			foreach (var pair in Directory.GetDirectories(supportPath).ToDictionary(x => x.Substring(supportPath.Length + 1)))
+				mods.Add(Pair.New(pair.Key, pair.Value));
+
+			foreach (var m in Directory.GetFiles(supportPath, "*.oramod"))
+				mods.Add(Pair.New(Path.GetFileNameWithoutExtension(m), m));
+
+			return mods;
+		}
+
+		static Dictionary<string, Manifest> GetInstalledMods()
+		{
+			var ret = new Dictionary<string, Manifest>();
+			foreach (var pair in GetCandidateMods())
+			{
+				IReadOnlyPackage package = null;
```

Comment:
If you feel like it, you could already factor out the body of this loop into a `LoadMod` method or so, which will be needed for mod installation down the line, anyway.